src/components/InlineNotification/InlineNotification.tsx

```diff
@@ -0,0 +1,129 @@
+  };
+} = {
+  brand: {
+    name: 'info',
+    title: 'info',
+  },
+  success: {
+    name: 'check-circle',
+    title: 'success',
+  },
+  warning: {
+    name: 'warning',
+    title: 'alert',
+  },
+  yield: {
+    name: 'error',
+    title: 'review',
+  },
+};
+
+type Variant = typeof VARIANTS[number];
+
+type Props = {
+  /**
+   * CSS class names that can be appended to the component for styling.
+   */
+  className?: string;
+  /**
+   * Indicates inactive state for the full width variant. Overrides variant prop.
+   * Only to be used with isFullWidth.
+   */
+  inactive?: boolean;
+  /**
+   * Toggles notification that fills the full width of its container.
+   * Used for CardWithNotification recipe.
+   */
+  isFullWidth?: boolean;
+  /**
+   * Toggles the stronger background variants.
+   */
+  isStrong?: boolean;
+  /**
+   * The text contents of the tag, nested inside the component, in addition to the icon.
+   */
+  text: React.ReactNode;
+  /**
+   * The score as a string to be appended after the text.
+   */
+  score?: string;
+  /**
+   * The color variant of the tag. New variants should be supported
+   * in the VARIANTS array and by its separate style in CSS file.
+   */
+  variant: Variant;
+};
+
+/**
+ * BETA: This component is still a work in progress and is subject to change.
+ *
+ * ```ts
+ * import {InlineNotification} from "@chanzuckerberg/eds";
+ * ```
+ *
+ * This component provides an inline banner accompanied with an icon for messaging users.
+ */
+export const InlineNotification = ({
+  className,
+  inactive,
+  isFullWidth,
+  isStrong,
+  score,
+  text,
+  variant,
+  ...other
+}: Props) => {
+  if (!isFullWidth && inactive && process.env.NODE_ENV !== 'production') {
+    throw new Error('inactive prop must be used with isFullWidth prop.');
+  }
+  const componentClassName = clsx(
+    styles['inline-notification'],
+    styles[`inline-notification--${variant}`],
+    !inactive && isStrong && styles[`inline-notification--${variant}-strong`],
+    isFullWidth && styles[`inline-notification--full-width`],
+    className,
+  );
+  const iconClassName = clsx(
+    styles['inline-notification__icon'],
+    styles[`inline-notification__icon--${variant}`],
+    inactive && styles[`inline-notification__icon--inactive`],
+  );
+  const textClassName = clsx(
+    styles[`inline-notification__text`],
+    inactive && styles[`inline-notification__text--inactive`],
+  );
+  return (
+    <div className={componentClassName} {...other}>
+      <Icon
+        className={iconClassName}
+        name={variantToIconAssetsMap[variant].name}
+        purpose="informative"
+        size="1.5rem"
+        title={variantToIconAssetsMap[variant].title}
+      />
+      <Text
+        as="span"
+        className={textClassName}
+        variant={variant === 'yield' ? 'base' : variant}
+      >
+        {text}
+        {score && (
+          <Text as="span" className={textClassName} variant="base">
```

Comment:
Not sure if extra element is necessary to make score 'base' variant but not sure how otherwise

Reply:
hmmm I'm not sure we should handle `score` in this component. `InlineNotification` should be pretty general-purpose / can accept anything in its container. If people want to pass in something like "Score", that should either:
- Be a recipe (i.e. they always do it themselves)
- Be a separate, more specific component (i.e. `ScoreNotification`) that can call `InlineNotification` and basically implement the recipe

Reply:
That makes sense to me, will remove `score` from the scope of the `InlineNotification` and use it as a recipe since the text allows for ReactNodes